lib/libxdp/xsk.c

```diff
@@ -898,14 +902,20 @@ int xsk_socket__create_shared(struct xsk_socket **xsk_ptr,
 		tx_setup_done = umem->tx_ring_setup_done;
 	}
 
-	ctx = xsk_get_ctx(umem, ifindex, queue_id);
+	optlen = sizeof(netns_cookie);
+	if (getsockopt(xsk->fd, SOL_SOCKET, SO_NETNS_COOKIE, &netns_cookie, &optlen) != 0) {
+		err = -errno;
+		goto out_socket;
+	}
```

Comment:
Given [this comment](https://github.com/xdp-project/xdp-tools/pull/205#issuecomment-1179375437), maybe we could conditionally compile this `getsockopt` call and instead set `netns_cookie` to 0 on kernels older than 5.14.

Question for maintainers: is use of the `KERNEL_VERSION` macro acceptable here?

Reply:
Nope, can't use `KERNEL_VERSION`, features may be backported. You can just detect the return value at runtime, though, and fall back to setting a fixed value. Something like this:

https://github.com/xdp-project/bpf-examples/commit/e92fd14fafb97603546e82814450d92396cd3636

Reply:
Thank you for the example. I've updated the PR based on this feedback (see commit 2c9573f88deb1587630285b3567f82894aa2a955).

Reply:
Alright, cool! Please squash the fixup into the original commit, though... :)

Reply:
squashed